dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,424 @@
+        final List<ServiceDescriptor> serviceDescriptors = serviceIntegrationAPI.getServiceDescriptors(user);
+        final List<Host> hosts = getHosts(user);
+        for (final ServiceDescriptor serviceDescriptor : serviceDescriptors) {
+            final String serviceKey = serviceDescriptor.getKey();
+            final long configurationsCount = computeSitesWithConfigurations(serviceKey, hosts, user).size();
+            views.add(new ServiceIntegrationView(serviceDescriptor, configurationsCount));
+        }
+        return views.stream().sorted(compareByNameAndConfiguration).collect(CollectionsUtils.toImmutableList());
+    }
+
+    /**
+     * This gets you a view composed of the service-key and all the hosts that have configurations
+     * @param serviceKey unique service id for the given host.
+     * @param user Logged in user
+     * @return view
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    Optional<ServiceIntegrationView> getServiceIntegrationSiteView(final String serviceKey,
+            final User user)
+            throws DotSecurityException, DotDataException {
+
+        final Optional<ServiceDescriptor> serviceDescriptorOptional = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (serviceDescriptorOptional.isPresent()) {
+            final ImmutableList.Builder<SiteView> hostViewBuilder = new ImmutableList.Builder<>();
+
+            final ServiceDescriptor serviceDescriptor = serviceDescriptorOptional.get();
+            final List<Host> hosts = getHosts(user);
+            final List<Host> hostsWithConfigurations = computeSitesWithConfigurations(serviceKey, hosts, user);
+            for (final Host host : hostsWithConfigurations) {
+                hostViewBuilder.add(new SiteView(host.getIdentifier(), host.getHostname()));
+            }
+            return Optional.of(
+                    new ServiceIntegrationView(serviceDescriptor, hostsWithConfigurations.size(), hostViewBuilder.build())
+            );
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * This gives you a detailed view with all the configuration secrets for a given service-key host pair.
+     * @param serviceKey unique service id for the given host.
+     * @param siteId Host Id
+     * @param user Logged in user
+     * @return view
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    Optional<ServiceIntegrationView> getServiceIntegrationSiteDetailedView(
+            final String serviceKey, final String siteId,
+            final User user)
+            throws DotSecurityException, DotDataException {
+        final Optional<ServiceDescriptor> serviceDescriptorOptional = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (serviceDescriptorOptional.isPresent()) {
+            final ServiceDescriptor serviceDescriptor = serviceDescriptorOptional.get();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)